src/contract.rs

```diff
@@ -0,0 +1,828 @@
+    .into_iter()
+    .find(|(current_bid_id, _)| *current_bid_id == bid_id.u64())
+    .map(|(_, bid)| bid).expect("REASON");// Construct and return BidResponse with the retrieved Bid
+    Ok(BidResponse { bid })
+}
+
+
+// Function to check the balance of an address for a given denom
+pub fn check_balance(deps: Deps, address: &str, denom: &str) -> StdResult<Uint128> {
+    let query_result = deps.querier.query_balance(address, denom);
+
+    // Handle the query result
+    match query_result {
+        Ok(balance) => Ok(balance.amount),
+        Err(err) => {
+            // Handle query error (e.g., address not found, denom not found, etc.)
+            Err(err)
+        }
+    }
+}
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use cosmwasm_std::testing::{mock_dependencies, mock_env, mock_info, MOCK_CONTRACT_ADDR,MockQuerier};
+    use cosmwasm_std::{from_binary, Addr, CosmosMsg, WasmMsg,attr,BankQuery};
+    use cosmwasm_std::BalanceResponse;
+    use cosmwasm_std::Querier;
+    
+    #[test]
+    fn test_create_deal() {
+      //  let mut deps = mock_dependencies();
+        let mut info = mock_info("sender", &[]);
+        let mut deps = mock_dependencies(); // Initialize with empty storage
+        
+        // Initialize sender and recipient addresses
+        let sender_address = Addr::unchecked("sender");
+        let fee_collector_address = Addr::unchecked("fee_addr");
+
+    // Initialize sender and fee collector balances
+        deps.querier.update_balance(sender_address.clone(), coins(100, "uotc"));
+        deps.querier.update_balance(fee_collector_address.clone(), coins(0, "uotc"));
+        let mut env = mock_env();
+        env.block.height = 0;
+        let value=info.sender.to_string();
+        let msg = InstantiateMsg {
+            admin: Some(value),
+            fee_collector: String::from("fee_addr"),
+            fee_denom: String::from("uotc"),
+            deal_creation_fee: 10,
+            fee_percent: Decimal::percent(1),
+        };
+        let res = instantiate(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+        assert_eq!(res.attributes,vec![attr("method", "instantiate"),
+        attr("owner","sender"),attr("addr","fee_addr")]);
+      
+        let balance=check_balance(deps.as_ref(), "sender", "uotc");
+      //  should create deal
+      let create_deal_msg = CreateDealMsg {
+        deal_creator: Addr::unchecked(info.sender.to_string()),
+        min_cap: Uint128::new(100),
+        total_bid: Uint128::new(0),
+        deal_token_denom: "token_denom".to_string(),
+        deal_token_amount: Uint128::new(1000),
+        start_block: Uint128::new(1000),
+        end_block: Uint128::new(2000),
+        bid_token_denom: "bid_token_denom".to_string(),
+        min_price: Decimal::from_ratio(0u128, 2u128),
+    };
+    let msg = ExecuteMsg::CreateDeal(create_deal_msg);
+    let res = execute(deps.as_mut(), mock_env(), info.clone(), msg.clone()).unwrap();
+
+    
+    println!("{:?}",res);
```

Comment:
Remove logs if not needed wherever present

Reply:
Sure